lib/broadside/target.rb

```diff
@@ -20,90 +23,55 @@ class Target
       :task_definition_config
     )
 
-    DEFAULT_INSTANCE = 0
+    validates :cluster, :docker_image, :name, presence: true
+    validates :scale, numericality: { only_integer: true }
 
-    def initialize(name, options = {})
-      @name = name
-      @config = options
-
-      @bootstrap_commands = @config[:bootstrap_commands]
-      @cluster = @config[:cluster]
-      @command = @config[:command]
-      _env_files = @config[:env_files] || @config[:env_file]
-      @env_files = _env_files ? [*_env_files] : nil
-      @env_vars = {}
-      @instance = DEFAULT_INSTANCE || @config[:instance]
-      @predeploy_commands = @config[:predeploy_commands]
-      @scale = @config[:scale]
-      @service_config = @config[:service_config]
-      @tag = @config[:tag]
-      @task_definition_config = @config[:task_definition_config]
-
-      validate!
+    validates_each :bootstrap_commands, :predeploy_commands, allow_nil: true do |record, attr, val|
+      record.errors.add(attr, 'must be an array of arrays') unless val.is_a?(Array) && val.all? { |v| v.is_a?(Array) }
     end
 
-    def load_env_vars!
-      @env_files.flatten.each do |env_path|
-        env_file = Pathname.new(env_path)
-
-        unless env_file.absolute?
-          dir = config.config_file.nil? ? Dir.pwd : Pathname.new(config.config_file).dirname
-          env_file = env_file.expand_path(dir)
-        end
-
-        raise ArgumentError, "Could not find env_file '#{env_file}'!" unless env_file.exist?
-
-        begin
-          @env_vars.merge!(Dotenv.load(env_file))
-        rescue Dotenv::FormatError => e
-          raise e.class, "Dotenv error: '#{e.message}' while parsing #{env_file}", e.backtrace
-        end
-      end
-
-      # convert env vars to format ecs expects
-      @env_vars = @env_vars.map { |k, v| { 'name' => k, 'value' => v } }
+    validates_each :service_config, :task_definition_config, allow_nil: true do |record, attr, val|
+      record.errors.add(attr, 'must be a hash') unless val.is_a?(Hash)
```

Comment:
i contemplated actually sending requests to a stubbed version of `register_task_definition` and `register_service`... because the stubs will validate the input.  dunno if it's actually worth it but if you end up doing the ELB feature or other hairier config stuff it might be, because some of those AWS hashes are kind of intense.